app/controllers/status_controller.rb

```diff
@@ -1,8 +1,40 @@
 # frozen_string_literal: true
 
+require 'date'
+
 class StatusController < ApplicationController
+  include TokenUser
+
   def index
     render 'index', layout: false
   end
 
+  # /whoami returns basic information about the request client from Conjur's
+  # perspective, including a role_id from a provided access token (if provided)
+  # and the origin IP address of the request.
+  #
+  # This is useful for debugging authentication with access tokens and configuring
+  # proxies or load balancers.
+  def whoami
+    render json: { 
+      client_ip: request.ip,
+      user_agent: request.user_agent,
+    }.merge(token_attributes)
+  end
+
+  def token_attributes
+    if token_user?
+      {
+        account: token_user.account,
+        login: token_user.login,
+        token_issued_at: Time.at(token_user.token.claims["iat"])
+      }
+    else
```

Comment:
Can you manage with `401` in case user doesn't have `token_user` - we can add error message if we need to distingish between `401` use cases

Randering and suppliing 200 OK (cache etc) seems less tight